### ISSUE TITLE
IDISP001 and IDISP004 false positives / add special case for MemoryStream

### ISSUE THREAD

**Issue by erichiller** [DotNetAnalyzers/IDisposableAnalyzers#541]
Use of `MemoryStream` triggers [`IDISP001`] and [`IDISP004`] however, while `MemoryStream` implements `IDisposable` it [doesn't actually have disposable resources](https://learn.microsoft.com/en-us/dotnet/api/system.io.memorystream?view=net-7.0#remarks).

`CA2000` has a [special case for Stream types because of this](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2000#special-cases). Their special cases are:
- `System.IO.Stream`
- `System.IO.StringReader`
- `System.IO.TextReader`
- `System.IO.TextWriter`
- `System.Resources.IResourceReader`

It would be good if these two analyzers had the same special cases. Alternatively (and a better, more future-proofed solution) would be to have a configuration rule, ideally something matching the EditorConfig rule like `CA2000`'s `excluded_symbol_names` and `excluded_type_names_with_derived_types`:
```editorconfig
dotnet_code_quality.IDISPxxx.excluded_symbol_names = MyType
dotnet_code_quality.IDISPxxx.excluded_type_names_with_derived_types = MyType
```

Example [`IDISP001`]:
```cs
var memoryStream = new MemoryStream( "{ }"u8.ToArray() );
```

Example [`IDISP004`]:
```cs
var memoryStream = config.AddJsonStream( new MemoryStream( "{ }"u8.ToArray() ) );
```

[`IDISP001`]:https://github.com/DotNetAnalyzers/IDisposableAnalyzers/blob/master/documentation/IDISP001.md
[`IDISP004`]:https://github.com/DotNetAnalyzers/IDisposableAnalyzers/blob/master/documentation/IDISP004.md
